avm/res/maps/account/main.bicep

```diff
@@ -0,0 +1,247 @@
+              ? 'userAssignedIdentity'
+              : 'systemAssignedIdentity'
+          }
+          keyEncryptionKeyUrl: customerManagedKey.?keyEncryptionKeyUrl
+        }
+      }
+      requireInfrastructureEncryption: requireInfrastructureEncryption
+    }
+  : {}
+
+var corsRulesProperty = [
+  for rule in corsRules ?? []: {
+    allowedOrigins: rule.allowedOrigins
+  }
+]
+
+var locationProperty = [
+  for dataLocation in locations ?? []: {
+    locationName: dataLocation
+  }
+]
+
+var properties = {
+  linkedResources: linkedResources ?? []
+  cors: {
+    corsRules: corsRulesProperty
+  }
+  disableLocalAuth: disableLocalAuth
+  locations: locationProperty
+}
+
+resource mapsAccount 'Microsoft.Maps/accounts@2024-07-01-preview' = {
+  name: name
+  location: location
+  sku: {
+    name: sku
+  }
+  identity: identity
+  kind: kind
+  tags: tags
+  properties: union(encryptionProperties, properties)
+}
+
+resource mapsAccount_roleAssignments 'Microsoft.Authorization/roleAssignments@2022-04-01' = [
+  for (roleAssignment, index) in (formattedRoleAssignments ?? []): {
+    name: roleAssignment.?name ?? guid(mapsAccount.id, roleAssignment.principalId, roleAssignment.roleDefinitionId)
+    properties: {
+      roleDefinitionId: roleAssignment.roleDefinitionId
+      principalId: roleAssignment.principalId
+      description: roleAssignment.?description
+      principalType: roleAssignment.?principalType
+      condition: roleAssignment.?condition
+      conditionVersion: !empty(roleAssignment.?condition) ? (roleAssignment.?conditionVersion ?? '2.0') : null // Must only be set if condtion is set
+      delegatedManagedIdentityResourceId: roleAssignment.?delegatedManagedIdentityResourceId
+    }
+    scope: mapsAccount
+  }
+]
+
+// ============ //
+// Outputs      //
+// ============ //
+@description('The resource ID of the Maps Account.')
+output resourceId string = mapsAccount.id
+
+@description('The name of the resource group the Maps Account was deployed into.')
+output resourceGroupName string = resourceGroup().name
+
+@description('The name of the Maps Account.')
+output name string = mapsAccount.name
+
+@description('The principal ID of the system assigned identity.')
+output systemAssignedMIPrincipalId string? = mapsAccount.?identity.?principalId
+
+@description('The location the resource was deployed into.')
+output location string = mapsAccount.location
+
+// ================ //
+// Definitions      //
+// ================ //
+@export()
+@description('The type of the CORS rule.')
+type corsRuleType = {
+  @description('Required. The allowed origins for the CORS rule.')
+  allowedOrigins: string[]
+}
+
+@export()
+@description('The type of the linked resource.')
+type linkedResourceType = {
+  @description('Required. ARM resource id in the form: \'/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Storage/accounts/{storageName}\'.')
+  id: string
```

Comment:
Should be updated to `resourceId` and then in the parameter usage be mapped to the `id` property via the `maps` function in accordance with [snfr22](https://azure.github.io/Azure-Verified-Modules/spec/SNFR22/#id-snfr22---category-inputs---parametersvariables-for-resource-ids).